geometry/include/pcl/geometry/mesh_base.h

```diff
@@ -894,7 +894,7 @@ class MeshBase {
   inline void
   resizeVertices(const std::size_t n, const VertexData& data = VertexData())
   {
-    vertices_.resize(n);
+    vertices_.resize(n, Vertex());
```

Comment:
Whats the reasoning about this change?

Reply:
When compiling as C++20, there is an error ([see here](https://dev.azure.com/PointCloudLibrary/pcl/_build/results?buildId=22118&view=logs&j=d5a8bb06-e546-5f6c-a052-332a9ca6ac18&t=1cb51625-174f-55bc-8807-1de378e39b6a&l=6024)). This is because the [DefaultInsertable](https://en.cppreference.com/w/cpp/named_req/DefaultInsertable) behaviour changed with C++20. These changes seemed like the least risky way to keep the old behaviour

Reply:
I can't reproduce it on windows. Wonder why it can't use/access the private constructor.

Reply:
Normally this error only appears under clang, because `resizeVertices` and the other two are not used in any tests and not compiled. Clang seems to even parse functions that are inline and not actually used anywhere, so it discovers more problems than gcc and msvc. I can trigger this error with gcc if I e.g. add `mesh.resizeVertices(3);` in [this line](https://github.com/PointCloudLibrary/pcl/blob/master/test/geometry/test_triangle_mesh.cpp#L129).

Reply:
Hmm. I get a complete different error - about resize on the DataCloudT::value type:
C:\PCL\pcl\geometry\include\pcl\geometry\mesh_base.h(2029): error C2228: left of '.resize' must have class/struct/union

Reply:
That could be a second error, I will look into that. But you don't see anything about line 897 or `std::construct_at`?

Reply:
Bot yet, will try with your latest push asap.

Reply:
It compiles now and run successfully. No errors about the construct_at.

Reply:
But as far as I could find out, there isn't much difference, so we can just continue.